src/Resources/js/modules/participation-response-handler.ts

```diff
@@ -14,13 +14,20 @@ interface SwapResponse extends ActionResponse {
 
 export interface ToggleResponse extends SwapResponse {
     participantsCount: number;
-    bookedDishes: string[];
+    bookedDishSlugs: string[];
 }
 
 
 export class ParticipationResponseHandler {
     public static onSuccessfulToggle($checkbox: JQuery, response: ToggleResponse) {
-        ParticipationUpdateHandler.toggleAction($checkbox, response);
+        const toggleData = {
```

Comment:
Well, basically it's the same... Anonymous object which gets interpreted as ToggleData. If you don't like the long parameter list (if it's too smelly for you), then please use ToggleResponse instead of this solution.

Reply:
No, this is not the same. We are not using anonymous object here. I can specify type like "const toggleData: ToggleData = {" here. But this is redundant as Typescript infer I believe from method signature. If this bothers you then I can specify the type.

Reply:
Can you specify a type for the const object? I tried but I can't without suppressing a warning. Well, I'm not a typescript expert. I see that if the parameter list matches then the interface is "valid" as a parameter. If you add "actionText2" to it, it's still valid, because most of the interface members are valid (actionText2 isn't and gets ignored). So I'm not sure if it's really a ToggleData object or if an anonymous object is interpreted as one (because of some matches in the members).

If the interfaces are the **same** (ToggleResponse and ToggleData) you don't need the const object at all. You can use:
```public static onSuccessfulToggle($checkbox: JQuery, response: ToggleResponse) {
        ParticipationUpdateHandler.toggleAction($checkbox, response);
    }
```
and in the other file:
```
public static toggleAction($checkbox: JQuery, test: ToggleData) {
```
and it will work too because typescript will handle "the cast" (or whatever) for you. 

Well, as I said, I'm not a typescipt expert and I don't know the best solution. 

Long parameter list --> put data to an object is a fine solution. It doesn't bother me. Do what you feel is the best solution! I just wanted to leave some ideas here.